src/matchbox/core.cljc

```diff
@@ -29,6 +29,18 @@
     [matchbox.registry :refer [register-listener register-auth-listener disable-auth-listener!]]
     #?(:cljs cljsjs.firebase)))
 
+;; macros
+(defmacro promise-> [& args]
```

Comment:
Interesting approach - we haven't leveraged the "new" promise return values in Matchbox yet.

Rather than creating our own (generic) promise sugar however, why not just use [something specialised](https://github.com/funcool/promesa)?

Existing callback error handling also needs work.. #40